services/jenkins/jenkins-coverage.service.js

```diff
@@ -1,80 +1,165 @@
-        const type = match[1] // c - cobertura | j - jacoco
-        const scheme = match[2] // http(s)
-        const host = match[3] // example.org:8080
-        const job = match[4] // folder/job
-        const format = match[5]
-        const options = {
-          json: true,
-          uri: `${scheme}://${host}/job/${job}/`,
-        }
-
-        if (job.indexOf('/') > -1) {
-          options.uri = `${scheme}://${host}/${job}/`
-        }
-
-        switch (type) {
-          case 'c':
-            options.uri +=
-              'lastBuild/cobertura/api/json?tree=results[elements[name,denominator,numerator,ratio]]'
-            break
-          case 'j':
-            options.uri +=
-              'lastBuild/jacoco/api/json?tree=instructionCoverage[covered,missed,percentage,total]'
-            break
-        }
-
-        if (serverSecrets && serverSecrets.jenkins_user) {
-          options.auth = {
-            user: serverSecrets.jenkins_user,
-            pass: serverSecrets.jenkins_pass,
-          }
-        }
-
-        const badgeData = getBadgeData('coverage', data)
-        request(options, (err, res, json) => {
-          if (checkErrorResponse(badgeData, err, res)) {
-            sendBadge(format, badgeData)
-            return
-          }
-
-          try {
-            const coverageObject = json.instructionCoverage
-            if (coverageObject === undefined) {
-              badgeData.text[1] = 'inaccessible'
-              sendBadge(format, badgeData)
-              return
-            }
-            const coverage = coverageObject.percentage
-            if (isNaN(coverage)) {
-              badgeData.text[1] = 'unknown'
-              sendBadge(format, badgeData)
-              return
-            }
-            badgeData.text[1] = coverage.toFixed(0) + '%'
-            badgeData.colorscheme = coveragePercentageColor(coverage)
-            sendBadge(format, badgeData)
-          } catch (e) {
-            badgeData.text[1] = 'invalid'
-            sendBadge(format, badgeData)
-          }
+const jacocoCoverageSchema = Joi.object({
+  instructionCoverage: Joi.object({
+    percentage: Joi.number()
+      .min(0)
+      .max(100)
+      .required(),
+  }).required(),
+}).required()
+
+const coberturaCoverageSchema = Joi.object({
+  results: Joi.object({
+    elements: Joi.array()
+      .items(
+        Joi.object({
+          name: 'Lines',
+          ratio: Joi.number()
+            .min(0)
+            .max(100)
+            .required(),
         })
-      })
+      )
+      .min(1)
+      .required(),
+  }).required(),
+}).required()
+
+class BaseJenkinsCoverage extends BaseJsonService {
+  async fetch({ url, options, schema }) {
+    return this._requestJson({
+      url,
+      options,
+      schema,
+      errorMessages: {
+        404: 'job or coverage not found',
+      },
+    })
+  }
+
+  static render({ coverage }) {
+    return {
+      message: coverage.toFixed(0) + '%',
```

Comment:
You can use any js expression in a template string, so this could be `${coverage.toFixed(0)}%` if you wanted.